beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingMetadataRepository.java

```diff
@@ -38,12 +38,20 @@
   @Query(value = "from HousekeepingMetadata t "
       + "where t.databaseName = :databaseName "
       + "and t.tableName = :tableName "
-      + "and (t.partitionName = :partitionName or t.partitionName is NULL) " // To handle special null case
+      + "and (t.partitionName = :partitionName or (:partitionName is NULL and t.partitionName is NULL)) " // To handle special null case
```

Comment:
:partitionName = "xyz"
t.partitionName = "xyz" or ("xyz" is null and t.partitionName is null)

:partitionName = null
t.partitionName = null or (null is null and t.partitionName is null)

I don't get it what problem are you trying to fix here?

Reply:
when we have `:partitionName = null` anything with that db and table name would get returned because `null is null` is always true; `t.partitionName = "xyz" or null is null`

which breaks the code because we only want 1 value back.

 So it needs to have that check to make sure that when you pass it null you only get the table entry back.

Reply:
Would splitting this up in two queries make it simpler in the code? on for getting the table record and one for getting the partition records?

Reply:
That is a discussion Vedant and I had when we found the problem, it was either this or have two queries. If you agree this is confusing then I think it is best to separate them out 👍

Reply:
it's ok to leave this seems to be the JPA way of doing it :)